pyanaconda/kickstart_dispatcher.py

```diff
@@ -0,0 +1,355 @@
+        :type sections: list
+        :param addons: names of addons to be returned or [] for all addons
+        :type addons: list
+
+        :return: list of filtered elements
+        :rtype: list(KickstartElement)
+        """
+
+        filtered_elements = []
+        for element in self._elements:
+            name = element.args[-1]
+            # addon
+            if element.args[0] == "%addon":
+                if addons is None:
+                    continue
+                if addons == [] or name in addons:
+                    filtered_elements.append(element)
+            # section
+            elif element.args[0].startswith("%"):
+                if sections is None:
+                    continue
+                if sections == [] or name in sections:
+                    filtered_elements.append(element)
+            # command
+            else:
+                if commands is None:
+                    continue
+                if commands == [] or name in commands:
+                    filtered_elements.append(element)
+        return filtered_elements
+
+    @staticmethod
+    def get_kickstart_from_elements(elements=None):
+        """Returns kickstart generated from elements.
+
+        :param elements: list of kickstart elements for generated kickstart
+        :type elements: list(KickstartElement)
+
+        :return: kickstart
+        :rtype: str
+        """
+        return "".join(element.content for element in elements)
+
+    def __str__(self):
+        return str(self._elements)
+
+
+class TrackedKickstartElements(KickstartElements):
+    """Container for kickstart elements with tracking."""
+    def __init__(self):
+        super().__init__()
+        self._processed_elements = set()
+
+    def get_elements(self, commands=None, sections=None, addons=None, process=False):
+        """Returns selected elements and tracks them as processed.
+
+        :param commands: names of commands to be returned or [] for all commands
+        :type commands: list
+        :param sections: names of sections to be returned or [] for all sections
+        :type sections: list
+        :param addons: names of addons to be returned or [] for all addons
+        :type addons: list
+        :param process: track returned elements as processed
+        :type process: bool
+
+        :return: list of filtered elements
+        :rtype: list(KickstartElement)
+        """
+        elements = super().get_elements(commands, sections, addons)
+        if process:
+            self._processed_elements.update(elements)
+        return elements
+
+    @property
+    def unprocessed_elements(self):
+        """List of all elements not tracked as processed.
+
+        :return: list of all elements not tracked as processed
+        :rtype: list(KickstartElement)
+        """
+        return [element for element in self._elements
+                if element not in self._processed_elements]
+
+
+class StoreSection(Section):
```

Comment:
This could be in separate file. It is not part using or handling `KickstartElements` class so why it should be here?

Reply:
I think i'll have one file for elements/tracking, another for kickstart parsing (including the section)